Can't construct a Rotation2d from a `units::turn_t` in C++

**Describe the bug**

The C++ `Rotation2d` constructor doesn't accept a `units::turn_t`. The compiler complains that the overloaded call is ambiguous.

```
build/temp.linux-x86_64-cpython-310/gensrc/wpimath_geometry/Rotation2d.cpp: In lambda function:
build/temp.linux-x86_64-cpython-310/gensrc/wpimath_geometry/Rotation2d.cpp:321:28: error: call of overloaded 'Rotation2d(units::angle::turn_t&)' is ambiguous
  321 |     return Rotation2d(value);
      |                            ^
In file included from /home/davo/dev/frc/robotpy/mostrobotpy/subprojects/robotpy-wpimath/wpimath/_impl/include/frc/geometry/Rotation2d.h:181,
                 from build/temp.linux-x86_64-cpython-310/gensrc/wpimath_geometry/Rotation2d.cpp:8:
/home/davo/dev/frc/robotpy/mostrobotpy/subprojects/robotpy-wpimath/wpimath/_impl/include/frc/geometry/Rotation2d.inc:19:11: note: candidate: 'constexpr frc::Rotation2d::Rotation2d(units::angle::degree_t)'
   19 | constexpr Rotation2d::Rotation2d(units::degree_t value)
      |           ^~~~~~~~~~
/home/davo/dev/frc/robotpy/mostrobotpy/subprojects/robotpy-wpimath/wpimath/_impl/include/frc/geometry/Rotation2d.inc:14:11: note: candidate: 'constexpr frc::Rotation2d::Rotation2d(units::angle::radian_t)'
   14 | constexpr Rotation2d::Rotation2d(units::radian_t value)
      |           ^~~~~~~~~~
```

**To Reproduce**

```cpp
frc::Rotation2d FromRotations(units::turn_t value) {
  return frc::Rotation2d(value);
}
```

**Expected behavior**

It should compile.

--- a/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
@@ -32,9 +32,7 @@ class WPILIB_DLLEXPORT Rotation2d {
    *
    * @param value The value of the angle.
    */
-  template <typename Angle>
-    requires units::traits::is_angle_unit_v<Angle>
-  constexpr Rotation2d(Angle value);  // NOLINT
+  constexpr Rotation2d(units::angle_unit auto value);  // NOLINT
 
   /**
    * Constructs a Rotation2d with the given x and y (cosine and sine)

--- a/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
@@ -28,18 +28,13 @@ class WPILIB_DLLEXPORT Rotation2d {
   constexpr Rotation2d() = default;
 
   /**
-   * Constructs a Rotation2d with the given radian value.
+   * Constructs a Rotation2d with the given angle.
    *
-   * @param value The value of the angle in radians.
+   * @param value The value of the angle.
    */
-  constexpr Rotation2d(units::radian_t value);  // NOLINT
-
-  /**
-   * Constructs a Rotation2d with the given degree value.
-   *
-   * @param value The value of the angle in degrees.
-   */
-  constexpr Rotation2d(units::degree_t value);  // NOLINT
+  template <typename Angle>
+    requires units::traits::is_angle_unit_v<Angle>
+  constexpr Rotation2d(Angle value);  // NOLINT
 
   /**
    * Constructs a Rotation2d with the given x and y (cosine and sine)

--- a/wpimath/src/main/native/include/frc/geometry/Rotation2d.inc
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation2d.inc
@@ -11,13 +11,12 @@
 
 namespace frc {
 
-constexpr Rotation2d::Rotation2d(units::radian_t value)
+template <typename Angle>
+  requires units::traits::is_angle_unit_v<Angle>
+constexpr Rotation2d::Rotation2d(Angle value)
     : m_value(value),
-      m_cos(gcem::cos(value.to<double>())),
-      m_sin(gcem::sin(value.to<double>())) {}
-
-constexpr Rotation2d::Rotation2d(units::degree_t value)
-    : Rotation2d(units::radian_t{value}) {}
+      m_cos(gcem::cos(value.template convert<units::radian>().value())),
+      m_sin(gcem::sin(value.template convert<units::radian>().value())) {}
 
 constexpr Rotation2d::Rotation2d(double x, double y) {
   double magnitude = gcem::hypot(x, y);

--- a/wpimath/src/main/native/include/frc/geometry/Rotation2d.inc
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation2d.inc
@@ -11,9 +11,7 @@
 
 namespace frc {
 
-template <typename Angle>
-  requires units::traits::is_angle_unit_v<Angle>
-constexpr Rotation2d::Rotation2d(Angle value)
+constexpr Rotation2d::Rotation2d(units::angle_unit auto value)
     : m_value(value),
       m_cos(gcem::cos(value.template convert<units::radian>().value())),
       m_sin(gcem::sin(value.template convert<units::radian>().value())) {}

--- a/wpimath/src/main/native/include/units/base.h
+++ b/wpimath/src/main/native/include/units/base.h
@@ -53,7 +53,6 @@
 
 #if !defined(_MSC_VER) || _MSC_VER > 1800
 #   define UNIT_HAS_LITERAL_SUPPORT
-#   define UNIT_HAS_VARIADIC_TEMPLATE_SUPPORT
 #endif
 
 #ifndef UNIT_LIB_DEFAULT_TYPE
@@ -358,25 +357,14 @@ template<> inline constexpr const char* abbreviation(const namespaceName::nameSi
 		/** @endcond */\
 	}
 
-#if defined(UNIT_HAS_VARIADIC_TEMPLATE_SUPPORT)
 #define UNIT_ADD_IS_UNIT_CATEGORY_TRAIT(unitCategory)\
 	namespace traits\
 	{\
 		template<typename... T> struct is_ ## unitCategory ## _unit : std::integral_constant<bool, units::all_true<units::traits::detail::is_ ## unitCategory ## _unit_impl<std::decay_t<T>>::value...>::value> {};\
 		template<typename... T> inline constexpr bool is_ ## unitCategory ## _unit_v = is_ ## unitCategory ## _unit<T...>::value;\
-	}
-#else
-#define UNIT_ADD_IS_UNIT_CATEGORY_TRAIT(unitCategory)\
-	namespace traits\
-	{\
-			template<typename T1, typename T2 = T1, typename T3 = T1>\
-			struct is_ ## unitCategory ## _unit : std::integral_constant<bool, units::traits::detail::is_ ## unitCategory ## _unit_impl<typename std::decay<T1>::type>::value &&\
-				units::traits::detail::is_ ## unitCategory ## _unit_impl<typename std::decay<T2>::type>::value &&\
-				units::traits::detail::is_ ## unitCategory ## _unit_impl<typename std::decay<T3>::type>::value>{};\
-			template<typename T1, typename T2 = T1, typename T3 = T1>\
-			inline constexpr bool is_ ## unitCategory ## _unit_v = is_ ## unitCategory ## _unit<T1, T2, T3>::value;\
-	}
-#endif
+	}\
+	template <typename T>\
+	concept unitCategory ## _unit = traits::is_ ## unitCategory ## _unit_v<T>;
 
 #define UNIT_ADD_CATEGORY_TRAIT(unitCategory)\
 	UNIT_ADD_CATEGORY_TRAIT_DETAIL(unitCategory)\

